libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TableFileStats.java

```diff
@@ -15,15 +19,39 @@ public class TableFileStats implements DataSource<FileStat> {
 
   @Override
   public Dataset<FileStat> get() {
-    return spark
-        .sql(String.format("SELECT file_path, file_size_in_bytes FROM %s.data_files", tableName))
-        .map(new FileStatMapper(), Encoders.bean(FileStat.class));
+    StructType fileSchema =
+        spark.sql(String.format("SELECT * FROM %s.data_files", tableName)).schema();
+    try {
+      fileSchema.apply("partition");
+      return spark
+          .sql(
+              String.format(
+                  "SELECT file_path, file_size_in_bytes, partition FROM %s.data_files", tableName))
+          .map(new FileStatMapper(), Encoders.bean(FileStat.class));
+    } catch (IllegalArgumentException e) {
+      return spark
+          .sql(
+              String.format(
+                  "SELECT file_path, file_size_in_bytes, null FROM %s.data_files", tableName))
+          .map(new FileStatMapper(), Encoders.bean(FileStat.class));
+    }
   }
 
   static class FileStatMapper implements MapFunction<Row, FileStat> {
     @Override
     public FileStat call(Row row) {
-      return FileStat.builder().path(row.getString(0)).size(row.getLong(1)).build();
+      List<String> partitionValues = new ArrayList<>();
+      Row partition = row.getStruct(2);
+      if (partition != null) {
+        for (int i = 0; i < partition.size(); i++) {
+          partitionValues.add(Objects.toString(partition.get(i)));
```

Comment:
Assuming the order here will match the order of partition columns, please confirm

Reply:
Yes. It will match.